lib/core/src/Cardano/Wallet/Api.hs

```diff
@@ -439,6 +443,16 @@ type GetAssetDefault = "wallets"
     :> Capture "policyId" (ApiT TokenPolicyId)
     :> Get '[JSON] ApiAsset
 
+-- | https://input-output-hk.github.io/cardano-wallet/api/#operation/mintBurnToken
+type MintBurnAsset n = "wallets"
+    :> Capture "walletId" (ApiT WalletId)
+    -- :> "assets"
+    -- :> "mint"
+    -- This has been changed to prevent overlapping routes:
+    :> "assets-mint"
```

Comment:
I'm not happy with this, does anyone know of a way to prevent servant from overlapping the routes?

The other routes in this namespace are of the form:

/assets/$policyId/...

So when it sees:

/assets/mint

It tries to treat "mint" as a policy ID and fails - even though this is a POST endpoint and the rest are GET.

Reply:
Hi @sevanspowell 

> It tries to treat "mint" as a policy ID and fails - even though this is a POST endpoint and the rest are GET.

Does the order in which the endpoints appear in the Servant type definition make a difference? What happens if you try the following order:
```hs
type Assets =
    ListAssets
    :<|> MintBurnAssets
    :<|> GetAsset
    :<|> GetAssetDefault
``

Reply:
Hey @jonathanknowles, the order in which the endpoints appear does not seem to matter in my experiments. I'll try and generate a failing test to better demonstrate what's going on.

Reply:
I've got a bit stuck trying to demonstrate this with a test, but I've been able to reproduce it by CURLing the running wallet.

The order of the Servant type definition makes no difference to the outcome.

Reply:
What I've done (suggested by @rvl), is to just make the mint/burn endpoint:

```
POST /wallets/.../assets
```

Reply:
The rdoc was complaining about duplicate keys in swagger.yaml. I've fixed it in 6d17d33.